app/models/service.rb

```diff
@@ -211,6 +209,14 @@ def power_states_match?(action)
     false
   end
 
+  def composite?
+    service_template ? service_template.service_type.to_s.include?('composite') : children.present?
```

Comment:
I like idea from @gmcculloug 

You can check if this object has a parent without a query.
But children checks that others are pointing to this object.
so it will always be a query hit.

and this will be an n+1